app/models/SessionRequestRepository.scala

```diff
@@ -0,0 +1,206 @@
+        "freeSlot" -> BSONDocument("$eq" -> false)))
+      case None      => Some(Json.obj("freeSlot" -> BSONDocument("$eq" -> false)))
+    }
+
+    collection
+      .flatMap(jsonCollection =>
+        jsonCollection.count(condition))
+  }
+
+  def getAllApprovedSession(implicit ex: ExecutionContext): Future[List[SessionRequestInfo]] = {
+    collection
+      .flatMap(jsonCollection =>
+        jsonCollection.
+          find(Json.obj("approved" -> true)).
+          cursor[SessionRequestInfo](ReadPreference.Primary)
+          .collect[List](-1, FailOnError[List[SessionRequestInfo]]()))
+  }
+
+  def approveSession(sessionId: String)(implicit ex: ExecutionContext): Future[WriteResult] = {
+    val selector = BSONDocument("_id" -> BSONDocument("$oid" -> sessionId))
+    val modifier = BSONDocument(
+      "$set" -> BSONDocument(
+        "approved" -> true,
+        "decline" -> false))
+
+    collection
+      .flatMap(jsonCollection =>
+        jsonCollection.update(selector, modifier))
+
+  }
+
+  def declineSession(sessionId: String)(implicit ex: ExecutionContext): Future[WriteResult] = {
+    val selector = BSONDocument("_id" -> BSONDocument("$oid" -> sessionId))
+    val modifier = BSONDocument(
+      "$set" -> BSONDocument(
+        "approved" -> false,
+        "decline" -> true))
+
+    collection
+      .flatMap(jsonCollection =>
+        jsonCollection.update(selector, modifier))
+
+  }
+
+  def updateDateForPendingSession(sessionId: String, date: BSONDateTime): Future[UpdateWriteResult] = {
+    val selector = BSONDocument("_id" -> BSONDocument("$oid" -> sessionId))
+    val modifier = BSONDocument(
+      "$set" -> BSONDocument(
+        "date" -> date))
+
+    collection
+      .flatMap(jsonCollection =>
+        jsonCollection.update(selector, modifier))
+  }
+
+  def getAllPendingSession: Future[List[SessionRequestInfo]] = {
+    val selector = BSONDocument("freeSlot" -> BSONDocument("$eq" -> false),
+      "approved" -> BSONDocument("$eq" -> false),
+      "decline" -> BSONDocument("$eq" -> false))
+
+    collection
+      .flatMap(jsonCollection =>
+        jsonCollection
+          .find(selector)
+          .cursor[SessionRequestInfo](ReadPreference.Primary)
+          .collect[List](-1, FailOnError[List[SessionRequestInfo]]()))
+
+  }
+
+  def deleteFreeSlot(id: String): Future[WriteResult] = {
+    val selector = BSONDocument("_id" -> BSONDocument("$oid" -> id))
+
+    collection
+      .flatMap(_.remove(selector))
+  }
+
+  def getAllFreeSlots: Future[List[SessionRequestInfo]] = {
+    val selector = BSONDocument("freeSlot" -> BSONDocument("$eq" -> true))
+
+    collection
+      .flatMap(jsonCollection =>
+        jsonCollection
+          .find(selector)
+          .cursor[SessionRequestInfo](ReadPreference.Primary)
+          .collect[List](-1, FailOnError[List[SessionRequestInfo]]()))
+  }
+
+  def getFreeSlotByDate(date: BSONDateTime): Future[Option[SessionRequestInfo]] = {
```

Comment:
Convert this to get free slot by id and keep id in dom element of the page it is used.